### PR TITLE
Upload documentation to "readthedocs.io"

### LINKED ISSUE #2333: Publish the docs to readthedocs.io
# Feature Request

**Is your feature request related to a problem?**

The current markdown docs in the GitHub repo might not be very user friendly / consumable.

**Describe the solution you'd like:**

Publish the docs to readthedocs.io, similar like what OpenTelemetry C++ is doing https://opentelemetry-cpp.readthedocs.io/en/latest/api/Overview.html.

### PR DESCRIPTION
**Pull request by mic-max**
Fixes #2333 

## Changes
* [x] Move all documentation from `/docs/*/README.md` to the `docs/public` folder
* [x] Choose an HTML theme - Match opentelemetry-cpp project?
* [x] Enable writing docs in markdown `.md` instead of `.rst`

image

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by mic-max** [comment]
I think I have a decent enough version for this for now: I added and refactored the `getting-started` projects for logs, tracing, and metrics. This is not complete so I will not complete the issue yet, but we can merge the current version and get feedback from team members.

There were also some suboptimal things that I haven't figured out to do in `sphinx` yet like importing the contents of a source file in our repository. See other TODO notes in the comments of the committed files.

File: docs/public/GettingHelp.md
Comment (reyang) [inline docs/public/GettingHelp.md:6]:
```suggestion
  [OpenTelemetry .NET GitHub repository](https://github.com/open-telemetry/opentelemetry-dotnet)
```

File: docs/public/GettingHelp.md
Comment (reyang) [inline docs/public/GettingHelp.md:7]:
```suggestion
  for further information and resources related to OpenTelemetry .NET.
```

Comment (reyang) [reply docs/public/GettingHelp.md:7]: There are more places that I didn't comment on.

File: docs/public/GettingHelp.md
Comment (reyang) [inline docs/public/GettingHelp.md:11]: I want to bring this to the @open-telemetry/dotnet-approvers' attention.
Maybe we need to be clear about the recommended place for questions.
I would think Slack channel is **worse** than GitHub issues, because GitHub issues are easier to search and tag, folks who came later can still benefit from that. Chat is something very transient.

File: docs/public/api/GettingStarted.rst
Comment (reyang) [inline docs/public/api/GettingStarted.rst:4]: suggest to order alphabetically:

```suggestion
All getting started projects start with the same setup: logs, metrics, traces.
```

File: docs/public/api/GettingStartedWithLogs.md
Comment (reyang) [inline docs/public/api/GettingStartedWithLogs.md:54]: Maybe a small script?

File: docs/public/api/GettingStartedWithMetrics.md
Comment (reyang) [inline docs/public/api/GettingStartedWithMetrics.md:5]:
```suggestion
* Counter
```

**Review by reyang — DISMISSED** [review]
LGTM in general, there are several minor things to be improved, please find my comments.

File: docs/public/GettingHelp.md
Comment (mic-max) [reply docs/public/GettingHelp.md:11]: I agree with the problem of a chat not being a good place to benefit others long term.
The `opentelemetry-cpp` project has [discussions open ](https://github.com/open-telemetry/opentelemetry-cpp/discussions)on their GitHub repository, so that could be another option for us to consider.

File: docs/public/GettingHelp.md
Comment (reyang) [reply docs/public/GettingHelp.md]: Reactive this since other places are not updated.

File: docs/public/api/GettingStartedWithLogs.md
Comment (mic-max) [reply docs/public/api/GettingStartedWithLogs.md:54]: Found out how to do this from Sphinx!
- https://www.jetbrains.com/pycharm/guide/tutorials/sphinx_sites/documentation/
- https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude

File: docs/public/GettingHelp.md
Comment (mic-max) [reply docs/public/GettingHelp.md]: Oh my bad, I hadn't pushed the changes. I'll hold off on resolving comments until I push my change in the future 👍🏻

File: docs/public/GettingHelp.md
Comment (cijothomas) [inline docs/public/GettingHelp.md:15]:
```suggestion
  [GitHub issue](https://github.com/open-telemetry/opentelemetry-dotnet/issues/new/choose).
```

File: README.md
Comment (cijothomas) [inline README.md:24]:
```suggestion
[Draft] As an application owner or a library author, you can find the getting started
```

File: docs/public/api/GettingStarted.rst
Comment (cijothomas) [inline docs/public/api/GettingStarted.rst:1]: lets move all of this to /sdk/.

There is not much to document in this repo for the API, as the APIs for all 3 signals are coming from .NET runtime repo.

File: docs/public/sdk/Metrics.md
Comment (pellared) [inline docs/public/sdk/Metrics.md:14]: ?

Comment (mic-max) [reply docs/public/sdk/Metrics.md:14]: I only included a setup for `Counter` for now and once we feel good about the format and how to publish these docs in a more user friendly format I can add the others. @reyang also just changed the metrics example, so I could replace this with the new version if there are no other concerns with this PR 🙂

Comment (pellared) [reply docs/public/sdk/Metrics.md:14]: I see two options for how we can reformat it.

1. We can add a descriptive sentence like: `The following steps demonstrate how to create a counter.`
1. We can change it to a header like: `## Create a counter`

**Comment by mic-max** [comment]
Going to make the change to the official opentelemetry.io website documentation instead

**Pull request closed without merging**
